compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -171,10 +172,8 @@ impl CoverageMapGenerator {
                     current_file_id += 1;
                 }
                 current_file_name = Some(file_name);
-                let c_filename = CString::new(file_name.to_string())
-                    .expect("null error converting filename to C string");
-                debug!("  file_id: {} = '{:?}'", current_file_id, c_filename);
-                let (filenames_index, _) = self.filenames.insert_full(c_filename);
```

Comment:
Notice that the old code here needs to convert every filename (`Symbol`) into a `CString` to insert it into the set of filenames, even though there will usually already be a duplicate entry (created by another function in the same source file).

Symbols are already naturally good for checking whether two strings are identical, so it makes sense to use symbols as set entries, and then only resolve them to strings later when we actually need to pass those strings through FFI.